src/Control/Email/Email.php

```diff
@@ -2,236 +2,181 @@
     }
 
     /**
      * Normalise email list from config merged with env vars
      *
-     * @param string $config Config key
-     * @param string $env Env variable key
-     * @return array Array of email addresses
+     * @return Address[]
      */
-    protected static function mergeConfiguredEmails($config, $env)
-    {
-        // Normalise config list
-        $normalised = [];
-        $source = (array)static::config()->get($config);
-        foreach ($source as $address => $name) {
-            if ($address && !is_numeric($address)) {
-                $normalised[$address] = $name;
-            } elseif ($name) {
-                $normalised[$name] = null;
-            }
+    private static function mergeConfiguredAddreses(string $configKey, string $envKey): array
+    {
+        $addresses = [];
+        $config = (array) static::config()->get($configKey);
+        $addresses = self::convertConfigToAddreses($config);
+        $env = Environment::getEnv($envKey);
+        if ($env) {
+            $addresses = array_merge($addresses, self::convertConfigToAddreses($env));
         }
-        $extra = Environment::getEnv($env);
-        if ($extra) {
-            $normalised[$extra] = null;
+        return $addresses;
+    }
+
+    private static function convertConfigToAddreses(array|string $config): array
+    {
+        $addresses = [];
+        if (is_array($config)) {
+            foreach ($config as $key => $val) {
+                if (filter_var($key, FILTER_VALIDATE_EMAIL)) {
+                    $addresses[] = new Address($key, $val);
+                } else {
+                    $addresses[] = new Address($val);
+                }
+            }
+        } else {
+            $addresses[] = new Address($config);
         }
-        return $normalised;
+        return $addresses;
     }
 
     /**
      * Encode an email-address to protect it from spambots.
      * At the moment only simple string substitutions,
      * which are not 100% safe from email harvesting.
      *
-     * @param string $email Email-address
-     * @param string $method Method for obfuscating/encoding the address
-     *    - 'direction': Reverse the text and then use CSS to put the text direction back to normal
-     *    - 'visible': Simple string substitution ('@' to '[at]', '.' to '[dot], '-' to [dash])
-     *    - 'hex': Hexadecimal URL-Encoding - useful for mailto: links
-     * @return string
+     * $method defines the method for obfuscating/encoding the address
+     * - 'direction': Reverse the text and then use CSS to put the text direction back to normal
+     * - 'visible': Simple string substitution ('@' to '[at]', '.' to '[dot], '-' to [dash])
+     * - 'hex': Hexadecimal URL-Encoding - useful for mailto: links
      */
-    public static function obfuscate($email, $method = 'visible')
+    public static function obfuscate(string $address, string $method = 'visible'): string
```

Comment:
Is the change to this (and others where you've done it) parameter name documented in the changelog? As of PHP 8.0 these are API so we need to make sure we document when we change them - and have a good reason for doing so.

Reply:
Reverted to $email